File: lib/cpp/example/HelloServer.cpp

#include <iostream>
#include <string>
#include "HelloMsg.h"
#include "rpc/RpcServer.h"
#include "rpc/NetEvent.h"
#include "rpc/EndPoint.h"

class  HelloMsgProxy :public shuai::opServiceProxyIF
{
 public:
	HelloMsgProxy(const rpc::Connection* conn=nullptr):opServiceProxyIF(conn){}

	virtual std::tuple<int,rpc::int8> login(const std::string& opendid) {
	  std::cout << opendid << std::endl;
	  return std::make_tuple(0,1); 
	};
	virtual std::tuple<int, shuai::user> xixi(shuai::user&  u) {
	  return std::make_tuple(0,u); 
	};
	virtual std::tuple<int, std::vector<shuai::role> >lala(std::map<rpc::int32, shuai::role>&  m, std::vector<rpc::int32> &  ai, std::vector<shuai::role> &  ar) {
	  std::vector<shuai::role> roles;
	  return std::make_tuple(0,roles); 
	};
	virtual std::tuple<int, rpc::int8>ping(rpc::int8 i8) {
	  return std::make_tuple(0,i8); 
	};

};
int main(int argc,char** argv)
{
	rpc::EndPoint ep(std::string(":2619"));
	using rpcServiceType = rpc::RpcService<rpc::TcpTransport, rpc::BinaryProtocol, HelloMsgProxy>;
	rpc::RpcServer<rpcServiceType> server(rpc::NetEvent::getInstance(),ep);
	if (-1 == server.open()) {
		std::cerr << "service.open " << std::endl;
	}

	int ret=rpc::NetEvent::getInstance()->eventLoop();
	server.close();
	return 0;
}

File: lib/go/example/server/HelloServer.go
package main

import (
	"kiss/rpc"
	"kiss/shuai"
)

func NewShuaiOpServiceProxyFactory(rpcService *rpc.RpcService) interface{} {
	return shuai.NewOpServiceProxy(rpcService, &shuaiOpServiceProxy{})
}

type shuaiOpServiceProxy struct {
}

func (this *shuaiOpServiceProxy) Login(openid string) (int8, error) {

	return 1, nil
}
func (this *shuaiOpServiceProxy) Ping(i8 int8) (int8, error) {

	return 0, nil
}
func (this *shuaiOpServiceProxy) Xixi(u shuai.User) (shuai.User, error) {
	return u, nil
}
func (this *shuaiOpServiceProxy) Lala(m map[int32]shuai.Role, ai []int32, ar []shuai.Role) ([]shuai.Role, error) {
	return ar, nil
}

func main() {
	serviceProxyFactory := []rpc.ServiceProxyFactory{NewShuaiOpServiceProxyFactory}

	factory := rpc.RpcServiecFactroy{TransFactory: rpc.TcpTransportFactory{},
		ProtoFactory: rpc.BinaryProtocolFactory{}, ServiceProxyFactory: serviceProxyFactory}

	event := rpc.NewNetEvent()
	ep := rpc.NewEndPoint("0.0.0.0:2619")
	server := rpc.NewRpcServer(event, *ep, factory)
	ret := server.Open()
	if ret == -1 {
		return
	}
	event.EventLoop()
	server.Close()
}
